### PR TITLE
ROC-5477 Prepopulate fields for GenerateOrder callback

### PR DESCRIPTION
**Pull request by simad**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5477

### Change description ###

Adding a about to start callback to the generate order event, to prepopulate the doc upload deadline, the eyewitness upload deadline and the direction list (all selected).

In other PRs I might move those callback-specific methods out of the ClaimService (which is doing too many things) into the CallbackService.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java [hunk 1]
```diff
@@ -41,4 +46,10 @@
     public String getValue() {
         return value;
     }
+
+    public static CaseEvent fromValue(String value) {
+        return Arrays.stream(values()).filter(event -> event.value.equals(value))
+            .findFirst()
+            .orElseThrow(() -> new BadRequestException("Unknown Case Event: " + value));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: Not sure "Bad Request" is semantically appropriate. At this level, perhaps `IllegalArgumentException` is better?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/CallbackService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public Callback getCallbackFor(String eventId, String callbackType) {
+        CaseEvent caseEvent = CaseEvent.fromValue(eventId);
+        
+        return Optional.ofNullable(callbacks.get(caseEvent))
+            .map(callbacks -> callbacks.get(callbackType))
+            .orElseThrow(() -> new BadRequestException(
+                format("Callback for event %s, type %s not implemented", eventId, callbackType)));
+    }
+
+    public interface Callback {
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/CallbackService.java]: Would like to see this explicitly annotated with `@FunctionalInterface` for clarity

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Mock
+    private Clock clock;
+
+    @Before
+    public void setUp() {
+        calculator = new LegalOrderGenerationDeadlinesCalculator(
+            clock,
+            workingDayIndicator);
+        when(clock.instant()).thenReturn(TODAY.atStartOfDay(ZoneId.systemDefault()).toInstant());
+        when(clock.getZone()).thenReturn(ZoneId.systemDefault());
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java]: Prooooobably okay, but makes me nervous, having worked a lot with timezones before. Can we make it explicitly fixed for all systems?

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        LocalDate responseDeadline = calculator.calculateOrderGenerationDeadlines();
+        LocalDate expected = TODAY.plusDays(DAYS_FOR_RESPONSE + DAYS_FOR_SERVICE);
+
+        assertThat(responseDeadline).isWeekday().isTheSame(expected);
+    }
+
+    @Test
+    public void shouldCalculateDeadlineAsTheNextWorkingDayIfDeadlineIsOnAHoliday() {
+        when(workingDayIndicator.isWorkingDay(any(LocalDate.class))).thenReturn(false).thenReturn(true);
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java]: `.thenReturn()` accepts varargs. Could be `.thenReturn(false, true)`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/CallbackService.java (hunk 2 above)
Comment (simad) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/CallbackService.java]: 👍

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java (hunk 3 above)
Comment (simad) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java]: 👍 I'll fix it (even tho it's a test) - good point

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java (hunk 4 above)
Comment (simad) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java]: 👍

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java (hunk 1 above)
Comment (simad) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/domain/CaseEvent.java]: True! Was just trying to mimick the logic which was already there - let's see how it looks like with a lower level exception

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java (hunk 4 above)
Comment (simad) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/LegalOrderGenerationDeadlinesCalculatorTest.java]: thank you, never remember the API for that :D

**Pull request merged**